eslint_temporary_suppressions.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * XXX: Temporary suppressions
+ *
+ * These rules are suppressed because we haven't yet fixed offending code.
+ *
+ * Want to help? Remove the suppression, fix any lint errors, and submit a PR.
+ */
+
+/** @type { import("eslint").Linter.Config[] } */
+export default [
+  /* Global rule suppressions */
+
+  {
+    rules: {
+      // Empty functions and blocks are useful (e.g `noop() {}`, `catch {}`) but can mask unintentionally omitted
+      // implementation. We should add explanatory comments like `// intentionally empty` and `// ignore error` in these
+      // scenarios to communicate intent.
+      'no-empty': 'off',
+      '@typescript-eslint/no-empty-function': 'off',
+
+      // We use these globals even though they're technically unstable features on our minimum
+      // supported node version.
+      'n/no-unsupported-features/node-builtins': [
+        'error',
+        {
+          ignores: [
+            'Blob',
+            'fetch',
+            'Headers',
+            'ReadableStream',
+            'Request',
+            'Response',
+            'stream.Readable.fromWeb',
+            'stream.Readable.toWeb',
+            'stream/web',
+          ],
+        },
+      ],
+
+      '@typescript-eslint/array-type': 'off',
```

Comment:
Technically a lot of these are auto-fixable, but I wanted to keep this PR as tightly-scoped to eslint changes as I could. Cleanup can be done in follow-ups.